src/ralph/accounts/models.py

```diff
@@ -179,6 +182,16 @@ def save(self, *args, **kwargs):
     def autocomplete_str(self):
         return '{} <i>{}</i>'.format(str(self), self.department)
 
+    @property
+    def permissions_hash(self):
+        """
+        Property used in template as a param to cache invalidation.
+        Hash for caching is calculated from user ID and its permissions.
+        """
+        perms_set = sorted(self.get_all_permissions())
```

Comment:
so first of all, use ''.join(perms_set) instead of urlqote - it's over 100 times faster:
```
x = user.get_all_permissions()

def test1():
    return ''.join(x)

def test2():
    return urlquote(x)

>>> timeit.timeit('test1()', setup='from __main__ import test1', number=10000)
... 0.10609826701693237
>>> timeit.timeit('test2()', setup='from __main__ import test2', number=10000)
... 15.055276021943428
```

Consider using frozenset instead of sorting:
```
perms_set = frozenset(self.get_all_permissions())
key = '{}:{}'.format(self.id, hash(perms_set))
```

It will be much faster than sorting and dumping whole perms.
```
def test_perms1():
    perms_set = frozenset(user.get_all_permissions())
    key = '{}:{}'.format(user.id, hash(perms_set))
    return hashlib.md5(force_bytes(key)).hexdigest()

def test_perms2():
    perms_set = sorted(user.get_all_permissions())
    key = '{}:{}'.format(user.id, ''.join(perms_set))
    return hashlib.md5(force_bytes(key)).hexdigest()

>>> timeit.timeit('test_perms1()', setup='from __main__ import test_perms1', number=10000)
... 1.1444226970197633
>>> timeit.timeit('test_perms2()', setup='from __main__ import test_perms2', number=10000)
... 3.365936833899468
```